### PR TITLE
[567500] Performances improvement

### PR DESCRIPTION
**Pull request by arnauddieumegard**

### CODE REVIEW

File: docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaLabelProviderHelper.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Obeo.
```
Comment (bzendagui) [inline docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaLabelProviderHelper.java:2]: The copyright is wrong. Please replace it with the Thalesgroup copyright

File: docgenhtml/plugins/org.polarsys.capella.docgen/templates/pattern._mGFhcAHoEemzNsJkc2kajg/method._mGFhcgHoEemzNsJkc2kajg.pt [hunk 2]
```diff
@@ -1 +1 @@
...
....eclipse.sirius.diagram.DDiagram org.eclipse.emf.common.util.EList org.eclipse.sirius.viewpoint.DRepresentationDescriptor org.polarsys.kitalpha.doc.gen.business.core.sirius.util.session.DiagramSessionHelper org.eclipse.sirius.business.api.dialect.DialectManager org.eclipse.sirius.viewpoint.description.DAnnotation org.polarsys.capella.core.diagram.helpers.DAnnotationHelper org.polarsys.capella.core.diagram.helpers.naming.DAnnotationSourceConstants org.eclipse.egf.common.helper.* java.util.* org.eclipse.emf.ecore.* org.eclipse.egf.model.pattern.* org.eclipse.egf.pattern.execution.* org.eclipse.egf.pattern.query.* org.eclipse.sirius.diagram.DSemanticDiagram org.polarsys.capella.core.data.capellacore.CapellaElement org.polarsys.capella.core.data.capellacore.NamedElement org.polarsys.capella.docgen.diagram.CapellaHelper org.polarsys.capella.docgen.util.DocGenHtmlCapellaControl org.polarsys.capella.docgen.util.DocGenHtmlCapellaUtil org.polarsys.capella.docgen.util.CapellaDiagramHelper org.polarsys.capella.docgen.util.CapellaServices org.polarsys.capella.docgen.util.CapellaLabelProviderHelper org.polarsys.kitalpha.doc.gen.business.core.preference.helper.DocgenDiagramPreferencesHelper"  %>
```
Comment (bzendagui) [inline docgenhtml/plugins/org.polarsys.capella.docgen/templates/pattern._mGFhcAHoEemzNsJkc2kajg/method._mGFhcgHoEemzNsJkc2kajg.pt:1]: No differences

Comment (arnauddieumegard) [reply docgenhtml/plugins/org.polarsys.capella.docgen/templates/pattern._mGFhcAHoEemzNsJkc2kajg/method._mGFhcgHoEemzNsJkc2kajg.pt:1]: There is a difference: LabelProviderHelper -> CapellaLabelProviderHelper

File: docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaLabelProviderHelper.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import org.eclipse.emf.common.notify.AdapterFactory;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.edit.provider.IItemLabelProvider;
+import org.polarsys.capella.core.model.handler.provider.CapellaAdapterFactoryProvider;
+import org.polarsys.kitalpha.doc.gen.business.core.util.LabelProviderHelper;
+
+public class CapellaLabelProviderHelper extends LabelProviderHelper {
+	
+	public static Object getIItemLabelProvider(EObject eObject) {
```
Comment (bzendagui) [inline docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaLabelProviderHelper.java:21]: No check on the type of eObject parameter is done. Does the Capella Adapter Factory able to adapt a no Capella element? 

May be you can add a check on the object type, if it is not a capella element, the API LabelProviderHelper.getText() can be used instead. BTW, this API can be used for Capella element and for any element available in the Capella Model.

File: docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaLabelProviderHelper.java (hunk 1 above)
Comment (arnauddieumegard) [reply docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaLabelProviderHelper.java]: Done

File: docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaLabelProviderHelper.java (hunk 3 above)
Comment (arnauddieumegard) [reply docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaLabelProviderHelper.java:21]: The Capella Adapter Factory is a composed adapter factory ending with an EMF ReflectiveItemProviderAdapterFactory so I would say there is no need for a Check to CapellaElement.

**Review by bzendagui — APPROVED** [review]
Ok for me

**Pull request merged**